scripts/dumps/bulk_genome_dump_from_core.py

```diff
@@ -0,0 +1,343 @@
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Wrapper of dump_genome_from_core.pl to dump a list of FASTA files
+"""
+import argparse
+import subprocess
+import os
+import sys
+import logging
+import yaml
+
+
+def setup_logging():
+    """
+    Sets up logging configuration.
+    """
+    logging.basicConfig(
+        level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+    )
+
+
+def detect_job_scheduler():
+    """
+    Detect if the system is using SLURM, LSF, or no job scheduler.
+
+    Returns:
+        str: The name of the detected job scheduler ('SLURM', 'LSF') or 'NONE' if no known scheduler is found.
+    """
+
+    schedulers = {
+        "SLURM": "srun",
+        "LSF": "bsub",
+    }
+
+    for scheduler, command in schedulers.items():
+        if (
+            subprocess.run(
+                ["which", command],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                check=False,
+            ).returncode
+            == 0
+        ):
+            return scheduler
+
+    return "NONE"  # Return 'NONE' if no scheduler is found
+
+
+def subprocess_call(
+    command,
+    stdout_file="/dev/null",
+    stderr_file="/dev/null",
+    work_dir=None,
+    shell=False,
+    use_job_scheduler=False,
+    job_name=None,
+):
+    """
+    Subprocess function to execute the given command line.
+
+    Args:
+        command (list): The command that the subprocess will execute.
+        work_dir (str): The location where the command should be run.
+        shell (bool): If True, the specified command will be executed through the shell.
+        use_job_scheduler (bool): If True, the command will be submitted to a job scheduler.
+
+    Returns:
+        str: The subprocess output or None otherwise.
+    """
+    if use_job_scheduler:
+        job_scheduler = detect_job_scheduler()
+
+        if job_scheduler == "NONE":
+            logging.error("No job scheduler detected.")
+            sys.exit(1)
+
+        if job_scheduler == "SLURM":
+            command = [
+                "sbatch",
+                "--time=1-00",
+                "--mem-per-cpu=4gb",
+                "--cpus-per-task=1",
+                "--export=ALL",
+                f"--output={stdout_file}",
+                f"--error={stderr_file}",
+                f"--job-name={job_name}",
+                f"--wrap={' '.join(command)}",
```

Comment:
If you `import shlex` at the top of the script, you could use `shlex.join` here (and in other, similar instances in the subsequent lines):
```suggestion
                f"--wrap={shlex.join(command)}",
```